File: codeforces/649_Div2/a.cpp
#include <iostream>
using namespace std;
int main()
{
    int t;
    cin>>t;
    while(t--)
    {
        int n,x,sum=0,temp,flag = 0;
        cin>>n>>x;
        for(int i=0;i<n;i++)
        {
            cin>>temp;
            sum+=temp;
            if(temp%x)
                flag = 1;
        }
        if(sum%x)
            printf("%d\n",n);
        else
        {
            if(flag)
                cout<<n-1<<endl;
            else
                cout<<-1<<endl;
        }
    }
}

File: codeforces/649_Div2/a.py
for _ in range(int(input())):
    n,x = map(int, input().split())
    l = list(map(int, input().split()))
    flag = 1
    for i in l:
        if(i%x):
            flag = 0
    if(flag):
        print(-1)
    else:
        if(sum(l)%x):
            print(n)
        else:
            j = 0
            while(l[j]%x==0):
                j += 1
            k = n-1
            while(l[k]%x == 0):
                k-=1
            print(max(n-j-1, k))
